completion/utilities.py

```diff
@@ -0,0 +1,21 @@
+#Python imports
+from __future__ import unicode_literals, absolute_import
+
+#Project imports
+from .exceptions import UnavailableCompletionData
+from .models import BlockCompletion
+from .waffle import visual_progress_enabled
+
+#Library imports
+
+
+def get_key_to_last_completed_course_block(user, course_id):
+    if not visual_progress_enabled(course_id):
```

Comment:
I don't like this check of a waffle flag here.  Why not have the caller of this function check if the waffle flag is on, and if it's not, just don't call the function.  Also, the exception raised here doesn't really match the condition that's raising it - just because a waffle switch is off doesn't mean that completion data doesn't exist.